server/src/main/java/org/apache/cassandra/sidecar/tasks/CassandraClusterSchema.java

```diff
@@ -0,0 +1,273 @@
+    private final InstanceMetadataFetcher instanceFetcher;
+    private final SecondBoundConfiguration tableSchemaRefreshTime;
+    private final CassandraBridgeFactory cassandraBridgeFactory;
+
+    public CassandraClusterSchema(InstanceMetadataFetcher instanceFetcher,
+                                  CdcDatabaseAccessor databaseAccessor,
+                                  SidecarConfiguration sidecarConfiguration,
+                                  CassandraBridgeFactory cassandraBridgeFactory)
+    {
+
+        this.instanceFetcher = instanceFetcher;
+        this.databaseAccessor = databaseAccessor;
+        this.sidecarConfiguration = sidecarConfiguration;
+        this.tableSchemaRefreshTime = sidecarConfiguration.serviceConfiguration().cdcConfiguration().tableSchemaRefreshTime();
+        this.cassandraBridgeFactory = cassandraBridgeFactory;
+    }
+
+    public void addSchemaChangeListener(Runnable listener)
+    {
+        schemaChangeListeners.add(listener);
+    }
+
+    public void refresh()
+    {
+        NodeSettings nodeSettings = instanceFetcher.callOnFirstAvailableInstance(instance-> instance.delegate().nodeSettings());
+        CassandraBridge cassandraBridge = cassandraBridgeFactory.get(nodeSettings.releaseVersion());
+        CdcBridge cdcBridge = CdcBridgeFactory.getCdcBridge(cassandraBridge);
+
+        try
+        {
+            LOGGER.info("Checking for schema changes...");
+            String fullSchemaText = databaseAccessor.fullSchema();
+            if (!fullSchemaText.equals(currSchemaText.get()))
+            {
+                LOGGER.info("Schema change detected, refreshing CDC tables");
+                currSchemaText.set(fullSchemaText);
+                Set<CqlTable> updatedCdcTables = buildCdcTables(fullSchemaText, databaseAccessor, tableIdCache, instanceFetcher, cassandraBridgeFactory);
+                LOGGER.info("Cdc enabled tables tables='{}'", 
+                            updatedCdcTables.stream()
+                                            .map(m -> String.format("%s.%s", m.keyspace(), m.table()))
+                                            .collect(Collectors.joining(",")));
+                cdcTables.set(updatedCdcTables);
+                cdcBridge.updateCdcSchema(updatedCdcTables, databaseAccessor.partitioner(),
+                                          ((keyspace, table) -> tableIdCache.get(TableIdentifier.of(keyspace, table))));
+                schemaChangeListeners.forEach(Runnable::run);
+            }
+        }
+        catch (IllegalStateException exception)
+        {
+            LOGGER.warn("There was a problem refreshing the schema. Database Accessor may not be ready", exception);
+            throw exception;
+        }
+        catch (Throwable t)
+        {
+            LOGGER.error("Unexpected error while refreshing the schema", t);
+            throw t;
+        }
+    }
+
+    public DurationSpec delay()
+    {
+        return tableSchemaRefreshTime;
```

Comment:
We should keep it like this for when we support dynamic configuration reloading in the future.
```suggestion
        return sidecarConfiguration.serviceConfiguration().cdcConfiguration().tableSchemaRefreshTime()
```